app/services/synchronize_source_service.rb

```diff
@@ -27,20 +26,18 @@ def synchronize_localization(loc)
     inc_loc.localization_key_is_deleted = localization_key_deleted?(loc)
     inc_loc.localization_key = find_localization_key(inc_loc)
     inc_loc.translated_value = loc['value']
+    return unless inc_loc.valid? # this also triggers connection with localization
```

Comment:
But `duplicated?` call also triggers connection with localization... I mean, there is a call to `set_localization` method right before any duplication checks